features/scripts/build_and_upload_app.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+require 'json'
+require 'net/http'
+
+def upload_app(username, access_key, app)
+  upload_uri = 'https://api-cloud.browserstack.com/app-automate/upload'
+  uri = URI(upload_uri)
+  request = Net::HTTP::Post.new(uri)
+  request.basic_auth(username, access_key)
+  request.set_form({ 'file' => File.new(app, 'rb') }, 'multipart/form-data')
+
+  puts "Uploading #{app} to #{upload_uri}"
+  res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+    http.request(request)
+  end
+
+  begin
+    body = res.body
+    response = JSON.parse body
+    raise "Upload failed due to error: #{response['error']}" if response.include?('error')
+    raise "Upload failed, response did not include and app_url: #{res}" unless response.include?('app_url')
+  rescue JSON::ParserError
+    raise "Error: expected JSON response, received: #{body}"
+  end
+
+  response['app_url']
+end
+
+# Build the test fixture apps
+`make test-fixtures`
+
+# Upload the IPA to BrowserStack
+url = upload_app ENV['BROWSER_STACK_USERNAME'], ENV['BROWSER_STACK_ACCESS_KEY'], 'features/fixtures/ios/output/iOSTestApp.ipa'
+puts "BrowserStack URL is #{url}"
+File.write('features/fixtures/ios/output/ipa_url.txt', url)
```

Comment:
Wouldn't it be simpler to use `curl`?

Reply:
Possibly, but we'd still need some script in there to process the response.  Ultimately, though, the repo should really have direct knowledge of BrowserStack at all - we should add an "upload app only" function to Maze Runner and abstract it away.  We could mark this section with a TODO against a ticket to refactor it?  Or, if it's too offensive, we can park this PR for a bit and add that function now?

Reply:
Now annotated with a ticket number to remove almost all of it.